tests/ga/test_extension.py

```diff
@@ -1540,86 +1545,92 @@ def test_ext_handler_no_reporting_status(self, *args):
         self._assert_handler_status(protocol.report_vm_status, "Ready", 1, "1.0.0")
         self._assert_ext_status(protocol.report_vm_status, ValidHandlerStatus.error, 0)
 
-    def test_wait_for_handler_completion_empty_exts(self, *args):
```

Comment:
Did a cleanup of existing tests for DependsOn scenario. There were some cases which are not possible in a real world scenario (eg: returning None for get_ext_handling_status) but they were still being tested due to excessive mocking. Modified the test cases to test scenarios that are more realisitc